byway/src/main/webapp/routepage.js

```diff
@@ -70,112 +75,117 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
     }
   });
 }
-
+ 
 /** Clear the stops panel in the html */
 function clearStops(){
   const stopList = document.getElementById('stop-list');
   if(stopList != null){
     stopList.innerText = "";
   }
 }
-
+ 
 /** Get the new list of stops from datastore onload */
 function getStopsOnload(){
   clearStops();
   fetch('/api/stop')
   .then(response => response.json())
   .then((stopsResponse) => {
-    stopsResponse.forEach((stop)=>{
-      stops.push(stop);
-    });
+    if(stopsResponse != null){
```

Comment:
i think the if statement can me omitted becase if stopsResponse is null, the foreach loop will just not run. I am not completely positive on that though, but may want to check

Reply:
I got an error in the web dev console when the JSON response was null, so I just included that check.